src/python_bindings/opt_to_py.cpp

```diff
@@ -0,0 +1,35 @@
+#include "opt_to_py.h"
```

Comment:
```suggestion
#include <functional>
#include <typeinfo>

#include "opt_to_py.h"
```
`typeinfo` for `typeid`, `functional` for `std::function` transitive includes are [not to be relied on](https://google.github.io/styleguide/cppguide.html#Include_What_You_Use).